### PR TITLE
core: add some tests around safety-speed triggering for ETCS/not-ETCS

### LINKED ISSUE #11777: core: add tests for ETCS corner cases
Continuation of https://github.com/OpenRailAssociation/osrd/pull/11694#discussion_r2088881793
> ETCS from 9500 to 10350: should keep the same behavior (no VISA)
> BAL all the way, but ETCS on 10100 to 10400: should have VISA ending on signal (10150) and on buffer-stop (10400)
> An initial stop at 100m under ETCS would be nice too to ensure "happy-path" behavior.

### PR DESCRIPTION
**Pull request by bougue-pe**
Follow-up on https://github.com/OpenRailAssociation/osrd/pull/11694#discussion_r2088881793
Fix https://github.com/OpenRailAssociation/osrd/issues/11777

### CODE REVIEW

**Review by louisgreiner — APPROVED** [review]
LGTM, the comments make it easy to follow, thanks

File: core/src/test/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulationTest.kt [hunk 1]
```diff
@@ -393,18 +393,178 @@ class StandaloneSimulationTest {
...
             )
 
-        val signalingRanges = buildSignalingRanges(infra, blocks.toIdxList(), chunkPath)
-        signalingRanges.put(9_500.meters, 10_400.meters, ETCS_LEVEL2.id)
-        val safetySpeedRanges =
-            makeSafetySpeedRanges(infra, chunkPath, routes.toIdxList(), schedule, signalingRanges)
-        val expected =
+        val signalingRangesBAL = buildSignalingRanges(infra, blocks.toIdxList(), chunkPath)
+
+        // ETCS_LEVEL2 range covers the first stop only: getting safetySpeed for the other stops
```
Comment (Erashin) [inline core/src/test/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulationTest.kt:398]: nit: I don't know how much clearer it would be if you mutualized this code, as you're doing it 5 times. Disregard if you find out it doesn't help.

**Review by Erashin — APPROVED** [review]

File: core/src/test/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulationTest.kt (hunk 1 above)
Comment (bougue-pe) [reply core/src/test/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulationTest.kt:398]: Indeed, we could shorten things a bit, on the `makeSafetySpeedRanges`, and a bit on signalingRanges modification and expected distanceRangeMap build.
It's not really worth the effort in my (lazy 😝) opinion, as DRY doesnt' apply as much on tests.
So I would not vote against this change, but I will skip the effort now 🙏
